pkg/metrics/series.go

```diff
@@ -221,6 +221,39 @@ func (a APIMetricType) seriesAPIV2Enum() int32 {
 // If a compressed payload is larger than the max, a new payload will be generated. This method returns a slice of
 // compressed protobuf marshaled MetricPayload objects.
 func (series Series) MarshalSplitCompress(bufferContext *marshaler.BufferContext) ([]*[]byte, error) {
+	return marshalSplitCompress(newSerieSliceIterator(series), bufferContext)
+}
+
+type serieIterator interface {
+	MoveNext() bool
+	Current() *Serie
+}
```

Comment:
An interface assertion would be useful here to indicate what this is implementing.

Reply:
Interface assertion added.